RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFMultiDepthClusterSoAProducer.cc

```diff
@@ -0,0 +1,182 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/InitialClusteringStepBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFClusterBuilderBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFClusterEnergyCorrectorBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/RecHitTopologicalCleanerBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/SeedFinderBase.h"
+#include "CondFormats/DataRecord/interface/HcalPFCutsRcd.h"
+#include "CondTools/Hcal/interface/HcalPFCutsHandler.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/Portable/interface/PortableObject.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/SynchronizingEDProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/MoveToDeviceCache.h"
+
+#include "RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFMultiDepthClusterParams.h"
+#include "RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFMultiDepthClusterizer_Alpaka.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFMultiDepthClusteringVarsDeviceCollection.h"
+
+#include "DataFormats/ParticleFlowReco/interface/alpaka/PFRecHitDeviceCollection.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  namespace {
+    using PFMultiDepthClusterParamsCache =
+        cms::alpakatools::MoveToDeviceCache<Device, PortableHostObject<PFMultiDepthClusterParams>>;
+  }
+
+  class PFMultiDepthClusterSoAProducer
+      : public stream::SynchronizingEDProducer<edm::GlobalCache<PFMultiDepthClusterParamsCache>> {
+  public:
+    PFMultiDepthClusterSoAProducer(edm::ParameterSet const& config, PFMultiDepthClusterParamsCache const*);
+
+    void acquire(device::Event const& event, device::EventSetup const&) override;
+    void produce(device::Event& event, device::EventSetup const&) override;
+    //
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+    static std::unique_ptr<PFMultiDepthClusterParamsCache> initializeGlobalCache(edm::ParameterSet const& config);
+    static void globalEndJob(PFMultiDepthClusterParamsCache*) {}
+
+  private:
+    const device::EDGetToken<reco::PFClusterDeviceCollection> inputPFClusterSoA_Token_;
+    const device::EDGetToken<reco::PFRecHitFractionDeviceCollection> inputPFRecHitFractionSoA_Token_;
+    const device::EDGetToken<reco::PFRecHitDeviceCollection> inputPFRecHitSoA_Token_;
+
+    const edm::EDGetTokenT<cms_uint32_t> inputPFClustersNum_Token_;
+    const edm::EDGetTokenT<cms_uint32_t> inputPFRecHitNum_Token_;
```

Comment:
These tokens do not seem to be used in the end. Can they be removed
```suggestion
```
or is this an indication of something being missing (I'd guess the former)?